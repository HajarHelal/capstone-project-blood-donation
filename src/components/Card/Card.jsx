import React from 'react';
import { PropTypes } from 'prop-types';

function Card({ cardImage, cardTitle, cardDesc, orientationText }) {
  const classesObjectStyles = {
    cardTitle: `text-primary-100 text-lg sm:text-xl md:text-2xl font-bold my-3 text-${orientationText}`,
    cardParagraph: `text-sm sm:text-base md:text-lg  text-${orientationText}`,
  };
  return (
    <div className="rounded">
      <div className="flex justify-center">
        <img className="w-16 h-12 sm:w-20 sm:h-16" src={cardImage} alt="icon" />
      </div>

      <div className="px-6 py-0 pb-6 my-5">
        <div className={classesObjectStyles.cardTitle}>{cardTitle}</div>
        <p className={classesObjectStyles.cardParagraph}>{cardDesc}</p>
      </div>
    </div>
  );
}
Card.propTypes = {
  cardImage: PropTypes.string,
  cardTitle: PropTypes.string,
  cardDesc: PropTypes.string.isRequired,
  orientationText: PropTypes.string,
};
export default Card;
