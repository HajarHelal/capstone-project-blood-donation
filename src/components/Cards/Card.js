import React from 'react';
import {  } from 'prop-types';

function Card({ cardImage, cardTitle, cardDesc }) {
  const classesObjectStyles = {
    cardProperties: 'bg-gray-200 max-w-sm my-5  shadow-lg ml-8 mt-15 border rounded',
    cardImagediv: 'w-32 mt-0 ml-32 lg:h',
    cardimageProperties: 'h-full w-full',
    cardTitle: 'text-orange-800 text-lg font-bold text-xl mb-2',
    cardParagraph: 'text-gray-700 text-lg',
  };

  return (
    <div>
      <div className="Card">
        <div data-testid="cardImage" className={classesObjectStyles.cardProperties}>
         <div className={classesObjectStyles.cardImagediv}> 
            <img
              className={classesObjectStyles.cardimageProperties}
              src={cardImage}
              alt="image"
            />
         </div>

          <div className="px-6 py-0 pb-1">
            <div data-testid="cardTitle" className={classesObjectStyles.cardTitle}>
              {cardTitle}
            </div>
            <p data-testid="cardDesc" className={classesObjectStyles.cardParagraph}>
              {cardDesc}
            </p>
          </div>
        </div>
      </div>
    </div>
  );
}
export default Card;
