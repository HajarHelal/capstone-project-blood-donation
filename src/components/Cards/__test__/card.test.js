import React from 'react';
import renderer from 'react-test-renderer';
import Card from '../../Card/Card';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Card
        cardImage="https://apifortress.com/doc/wp-content/uploads/2019/04/ForgeIcon_Black.svg"
        cardTitle=" Title "
        cardDesc=" Description "
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
