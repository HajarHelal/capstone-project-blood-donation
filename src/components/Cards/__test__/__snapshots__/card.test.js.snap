// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="Card"
  >
    <div
      className="max-w-sm  shadow-lg ml-8 mt-15 border rounded"
      data-testid="cardImage"
    >
      <section
        className="hero container max-w-screen-lg mx-auto"
      >
        <img
          alt="screenshot"
          className="mx-auto mt-8"
          src="https://apifortress.com/doc/wp-content/uploads/2019/04/ForgeIcon_Black.svg"
        />
      </section>
      <div
        className="px-6 py-4"
      >
        <div
          className="text-orange-800  font-bold text-xl mb-2"
          data-testid="cardTitle"
        >
           Title 
        </div>
        <p
          className="text-gray-700 text-lg"
          data-testid="cardDesc"
        >
           Description 
        </p>
      </div>
    </div>
  </div>
</div>
`;
